<a href="https://colab.research.google.com/github/FelipeDamicoCapitao/Trabalho_individual_Modulo4/blob/main/Trabalho_Individual_modulo4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importação Pandas

In [ ]:
import pandas as pd
import numpy as np

# Tabela Base


In [ ]:
colunas_dic = {'Dia': ["Segunda","Terça","Quarta","Quinta","Sexta","Sabado","Domingo"],
  'Horas Trabalhadas': [6,7,8,6,7,5,4],
  'Bugs Corrigidos': [3,2,1,4,3,2,1],
  'Tarefas Concluídas': [5,4,6,4,5,3,2]}

relatorio_dt= pd.DataFrame(colunas_dic)
relatorio_dt



Dia  Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas
0  Segunda                  6                3                   5
1    Terça                  7                2                   4
2   Quarta                  8                1                   6
3   Quinta                  6                4                   4
4    Sexta                  7                3                   5
5   Sabado                  5                2                   3
6  Domingo                  4                1                   2

In [ ]:
# Correlação
relatorio_dt.corr()

<ipython-input-68-9823896f953d>:2: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  relatorio_dt.corr()


Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas
Horas Trabalhadas            1.000000         0.079536            0.907895
Bugs Corrigidos              0.079536         1.000000            0.190885
Tarefas Concluídas           0.907895         0.190885            1.000000

# Tabela Horas Trabalhadas

In [ ]:
#Cria Tabela copiando duas colunas da Tabela Base
horas_df = relatorio_dt[["Dia","Horas Trabalhadas"]].copy()

#Adiciona 2 Colunas, Soma Acumulada e média Acumulada:
horas_df['Horas Acumuladas'] = horas_df['Horas Trabalhadas'].cumsum()
horas_df['Média Acumulada de Horas'] = round(horas_df['Horas Acumuladas'] / ( horas_df.index + 1),2)
horas_df



Dia  Horas Trabalhadas  Horas Acumuladas  Média Acumulada de Horas
0  Segunda                  6                 6                      6.00
1    Terça                  7                13                      6.50
2   Quarta                  8                21                      7.00
3   Quinta                  6                27                      6.75
4    Sexta                  7                34                      6.80
5   Sabado                  5                39                      6.50
6  Domingo                  4                43                      6.14

In [ ]:
media = horas_df["Horas Trabalhadas"].mean()
print("Media:", round(media, 2))

desvio = horas_df["Horas Trabalhadas"].std()
print("Desvio Padrão:", round(desvio,2))
print("Porcentagem Desvio/media [%]:", round(desvio/media*100,2) )

limite2= media + desvio
limite1= media - desvio

print("Dias que estão fora do desvio padrão:")
#Imprimir dias que tão fora do dia padrao com LOC
tab_aux = horas_df[["Dia", "Horas Trabalhadas"]].copy()
selecao = tab_aux.loc[(limite2 < horas_df["Horas Trabalhadas"]) | (limite1 > horas_df["Horas Trabalhadas"])]
display(selecao)


Media: 6.14
Desvio Padrão: 1.35
Porcentagem Desvio/media [%]: 21.9
Dias que estão fora do desvio padrão:


Dia  Horas Trabalhadas
2   Quarta                  8
6  Domingo                  4

# Tabela Bugs Corrigidos

In [ ]:
#Cria Tabela copiando duas colunas da Tabela Base
bugs_df = relatorio_dt[["Dia",'Horas Trabalhadas',"Bugs Corrigidos"]].copy()

#Adiciona 2 Colunas, Soma Acumulada e média Acumulada:
bugs_df['Bugs Corrigidos Acumulados'] = bugs_df['Bugs Corrigidos'].cumsum()
bugs_df['Média Acumulada'] = round(bugs_df['Bugs Corrigidos Acumulados'] / ( bugs_df.index + 1),2)


#Adiciona 1 COLUNA: BUGS/HORA TRABALHADA
bugs_df['Bugs Corrigidos por hora'] =  round(bugs_df['Bugs Corrigidos']/bugs_df['Horas Trabalhadas'],2)
bugs_df

Dia  Horas Trabalhadas  Bugs Corrigidos  Bugs Corrigidos Acumulados  \
0  Segunda                  6                3                           3   
1    Terça                  7                2                           5   
2   Quarta                  8                1                           6   
3   Quinta                  6                4                          10   
4    Sexta                  7                3                          13   
5   Sabado                  5                2                          15   
6  Domingo                  4                1                          16   

   Média Acumulada  Bugs Corrigidos por hora  
0             3.00                      0.50  
1             2.50                      0.29  
2             2.00                      0.12  
3             2.50                      0.67  
4             2.60                      0.43  
5             2.50                      0.40  
6             2.29                      0.25

In [ ]:
media = bugs_df['Bugs Corrigidos por hora'].mean()
print("Media:", round(media, 2))


desvio = bugs_df['Bugs Corrigidos por hora'].std()
print("Desvio Padrão:", round(desvio,2))

print("Porcentagem Desvio/media [%]:", round(desvio/media*100,2) )

limite2= media + desvio
limite1= media - desvio

print("Dias que estão fora do desvio padrão:")
#Imprimir dias que tão fora do dia padrao com LOC
tab_aux = bugs_df[["Dia", 'Bugs Corrigidos por hora']].copy()
selecao = tab_aux.loc[(limite2 < bugs_df['Bugs Corrigidos por hora']) | (limite1 > bugs_df['Bugs Corrigidos por hora'])]
display(selecao)


Media: 0.38
Desvio Padrão: 0.18
Porcentagem Desvio/media [%]: 47.32
Dias que estão fora do desvio padrão:


Dia  Bugs Corrigidos por hora
2  Quarta                      0.12
3  Quinta                      0.67

# Tabela Tarefas Concluídas

In [ ]:
#Cria Tabela copiando duas colunas da Tabela Base
tarefas_df = relatorio_dt[["Dia",'Horas Trabalhadas',"Tarefas Concluídas"]].copy()

#Adiciona 2 Colunas, Soma Acumulada e média Acumulada:
tarefas_df['Tarefas Acumuladas'] = tarefas_df['Tarefas Concluídas'].cumsum()
tarefas_df['Média Acumulada de Tarefas'] = round(tarefas_df['Tarefas Acumuladas'] / ( tarefas_df.index + 1),2)



#Adiciona 1 COLUNA: Tarefas/HORA TRABALHADA
tarefas_df['Tarefas Concluídas por hora'] =  round(tarefas_df['Tarefas Concluídas']/tarefas_df['Horas Trabalhadas'],2)
tarefas_df


Dia  Horas Trabalhadas  Tarefas Concluídas  Tarefas Acumuladas  \
0  Segunda                  6                   5                   5   
1    Terça                  7                   4                   9   
2   Quarta                  8                   6                  15   
3   Quinta                  6                   4                  19   
4    Sexta                  7                   5                  24   
5   Sabado                  5                   3                  27   
6  Domingo                  4                   2                  29   

   Média Acumulada de Tarefas  Tarefas Concluídas por hora  
0                        5.00                         0.83  
1                        4.50                         0.57  
2                        5.00                         0.75  
3                        4.75                         0.67  
4                        4.80                         0.71  
5                        4.50                         0.60  
6                        4.14                         0.50

In [ ]:
media = tarefas_df['Tarefas Concluídas por hora'].mean()
print("Media:", round(media, 2))

desvio = tarefas_df['Tarefas Concluídas por hora'].std()
print("Desvio Padrão:", round(desvio,2))

print("Porcentagem Desvio/media [%]:", round(desvio/media*100,2) )

limite2= media + desvio
limite1= media - desvio

print("Dias que estão fora do desvio padrão:")
#Imprimir dias que tão fora do dia padrao com LOC
tab_aux = tarefas_df[["Dia", 'Tarefas Concluídas por hora']].copy()
selecao = tab_aux.loc[(limite2 < tarefas_df['Tarefas Concluídas por hora']) | (limite1 > tarefas_df['Tarefas Concluídas por hora'])]
display(selecao)

Media: 0.66
Desvio Padrão: 0.11
Porcentagem Desvio/media [%]: 17.11
Dias que estão fora do desvio padrão:


Dia  Tarefas Concluídas por hora
0  Segunda                         0.83
6  Domingo                         0.50

# Eventos (Bugs+Tarefas)

In [ ]:
#Cria Tabela copiando duas colunas da Tabela Base
eventos_df = relatorio_dt[["Dia",'Horas Trabalhadas']].copy()
eventos_df["Eventos"] = relatorio_dt["Tarefas Concluídas"]+relatorio_dt["Bugs Corrigidos"]
eventos_df

#Adiciona 1 COLUNA: Eventos/HORA TRABALHADA
eventos_df['Eventos Realizados por hora'] =  round(eventos_df["Eventos"]/eventos_df['Horas Trabalhadas'],2)
eventos_df

Dia  Horas Trabalhadas  Eventos  Eventos Realizados por hora
0  Segunda                  6        8                         1.33
1    Terça                  7        6                         0.86
2   Quarta                  8        7                         0.88
3   Quinta                  6        8                         1.33
4    Sexta                  7        8                         1.14
5   Sabado                  5        5                         1.00
6  Domingo                  4        3                         0.75

# Análise

1 - Podemos perceber que há uma alta correlação entre a quantidade de horas trabalhadas e tarefas concluidas, o que indica que a quantidade de horas trabalhadas é afetada majoritariamente pelas quantidade de tarefas. Já a correlação entre bugs corrigidos e horas trabalhadas não é significativa, o que indica que esses nao interferem significavamente no horario empregado.

2 - O a quantidade de bugs varia muito dentro da semana,a razão entre o Desvio padrão e sua media é 47.32 %. Já a de tarefas é de 17,11%, proximo a de horas: 21,9%. Isso indica que a correção de bugs pode variar muito em quantidade durante ao longo da semana, apesar nao afetar tanto a quantidade de horas de trabalho empregadas

3 - Analisando dia com mais horas trabalhadas temos o dia com mais Tarefas concluidas e o segundo com maior média por hora, mas uma quantidade de Eventos(tarefa+bugs) por hora baixa, o que reforça que as tarefas são diretamente proporcionais às horas trabalhadas e os bugs corrigidos pouco afentam as horas trabalhadas

4 - Analisando o dia com menos horas trabahadas temos o menor numero de tarefas concluidas e a menor média por hora, refoçando a ligação entre quantidade de tarefas e horas trabalhadas.